src/omnicore/omnicore.cpp

```diff
@@ -2983,87 +2983,55 @@ void CMPTxList::LoadActivations()
     delete it;
 }
 
-int CMPTxList::setLastAlert(int blockHeight)
+void CMPTxList::LoadAlerts(int blockHeight)
 {
-    if (blockHeight > GetHeight()) blockHeight = GetHeight();
-    if (!pdb) return 0;
+    if (!pdb) return;
     Slice skey, svalue;
     Iterator* it = NewIterator();
-    string lastAlertTxid;
-    string lastAlertData;
-    int64_t lastAlertBlock = 0;
-    for(it->SeekToFirst(); it->Valid(); it->Next())
-    {
-       skey = it->key();
-       svalue = it->value();
-       string itData = svalue.ToString();
-       std::vector<std::string> vstr;
-       boost::split(vstr, itData, boost::is_any_of(":"), token_compress_on);
-       // we expect 5 tokens
-       if (4 == vstr.size())
-       {
-           if (atoi(vstr[2]) == OMNICORE_MESSAGE_TYPE_ALERT) // is it an alert?
-           {
-               if (atoi(vstr[0]) == 1) // is it valid?
-               {
-                    if ((atoi(vstr[1]) > lastAlertBlock) && (atoi(vstr[1]) <= blockHeight)) // is it the most recent and within our parsed range?
-                    {
-                        lastAlertTxid = skey.ToString();
-                        lastAlertData = svalue.ToString();
-                        lastAlertBlock = atoi(vstr[1]);
-                    }
-               }
-           }
-       }
-    }
-    delete it;
 
-    // if lastAlertTxid is not empty, load the alert and see if it's still valid - if so, copy to global_alert_message
-    if(lastAlertTxid.empty())
-    {
-        PrintToLog("DEBUG ALERT No alerts found to load\n");
-    }
-    else
-    {
-        PrintToLog("DEBUG ALERT Loading lastAlertTxid %s\n", lastAlertTxid);
-
-        // reparse lastAlertTxid
-        uint256 hash;
-        hash.SetHex(lastAlertTxid);
-        CTransaction wtx;
+    for (it->SeekToFirst(); it->Valid(); it->Next()) {
+        std::string itData = it->value().ToString();
+        std::vector<std::string> vstr;
+        boost::split(vstr, itData, boost::is_any_of(":"), token_compress_on);
+        if (4 != vstr.size()) continue; // unexpected number of tokens
+        if (atoi(vstr[2]) != OMNICORE_MESSAGE_TYPE_ALERT || atoi(vstr[0]) != 1) continue; // not a valid alert
+        uint256 txid = 0;
```

Comment:
Nit: there is no need to initialize `uint256` with 0 (which happens on it's own). I mention this, because `uint256` changes a bit in 0.11 and this line wouldn't be possible anymore.